### PR TITLE
go_1_21/buildGo121Module: remove

### PR DESCRIPTION
**Pull request by katexochen**
## Description of changes

Go 1.21 is EOL since 5 weeks and should be removed for the next release.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by katexochen** [comment]
Result of `nixpkgs-review` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package blacklisted:</summary>
  <ul>
    <li>nixos-install-tools</li>
  </ul>
</details>
<details>
  <summary>2 packages failed to build:</summary>
  <ul>
    <li>bazel-watcher</li>
    <li>zincsearch</li>
  </ul>
</details>
<details>
  <summary>6 packages built:</summary>
  <ul>
    <li>datadog-agent</li>
    <li>datadog-process-agent</li>
    <li>nomad_1_6</li>
    <li>sensu-go-agent</li>
    <li>sensu-go-backend</li>
    <li>sensu-go-cli</li>
  </ul>
</details>

@kalbasit for bazel-watcher
@dit7ya for zincsearch

File: pkgs/applications/networking/cluster/nomad/default.nix [hunk 1]
```diff
@@ -62,7 +61,7 @@ rec {
   nomad_1_5 = throwUnsupportaed "nomad_1_5";
 
   nomad_1_6 = generic {
-    buildGoModule = buildGo121Module;
```
Comment (techknowlogick) [inline pkgs/applications/networking/cluster/nomad/default.nix:65]: Would you be able to remove nomad 1.6 entirely instead?

Comment (katexochen) [reply pkgs/applications/networking/cluster/nomad/default.nix:65]: Sure!

Comment (katexochen) [reply pkgs/applications/networking/cluster/nomad/default.nix:65]: Done.

**Comment by katexochen** [comment]
@malt3 sent a patch to build `bazel-watcher` with `buildGoModule` instead with the upstream bazel files, removing any issues with upgrading Go. Thanks!

**Comment by katexochen** [comment]
Marking `zincsearch` broken for now as maintainer hasn't responded yet and I can't figure it out, opened an upstream issue.

File: pkgs/development/compilers/go/1.21.nix [hunk 2]
```diff
@@ -1,189 +0,0 @@
...
-      inherit mailcap;
-    })
-    # prepend the nix path to the zoneinfo files but also leave the original value for static binaries
-    # that run outside a nix server
-    (substituteAll {
-      src = ./tzdata-1.19.patch;
-      inherit tzdata;
-    })
-    ./remove-tools-1.11.patch
-    ./go_no_vendor_checks-1.21.patch
```
Comment (zowoq) [inline pkgs/development/compilers/go/1.21.nix:89]: This patch file can be removed.

Comment (katexochen) [reply pkgs/development/compilers/go/1.21.nix:89]: Done. :)

**Comment by Mic92** [comment]
Result of `nixpkgs-review pr 344622` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package marked as broken and skipped:</summary>
  <ul>
    <li>zincsearch</li>
  </ul>
</details>
<details>
  <summary>1 package blacklisted:</summary>
  <ul>
    <li>nixos-install-tools</li>
  </ul>
</details>
<details>
  <summary>6 packages built:</summary>
  <ul>
    <li>bazel-watcher</li>
    <li>datadog-agent</li>
    <li>datadog-process-agent</li>
    <li>sensu-go-agent</li>
    <li>sensu-go-backend</li>
    <li>sensu-go-cli</li>
  </ul>
</details>

**Review by zowoq — APPROVED** [review]

**Comment by ryanrasti** [comment]
Heads up that this PR breaks datadog-agent as re-introduces the issue in https://github.com/NixOS/nixpkgs/issues/299469

**Comment by Mic92** [comment]
If we don't find a fix for that we should maybe mark datadog-agent as broken?

**Pull request merged**